types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -336,7 +336,8 @@ static void xdg_surface_handle_surface_commit(struct wl_listener *listener,
 		return;
 	}
 
-	if (surface->role == WLR_XDG_SURFACE_ROLE_NONE) {
+	// surface->role might be NONE for inert popups
+	if (surface->surface->role == WLR_XDG_SURFACE_ROLE_NONE) {
```

Comment:
`surface->surface` is the `wlr_surface`, so it has nothing to do with xdg-shell. `surface->surface->role` is a `struct wlr_surface_role *` and not an enum, so we need to check whether it's NULL.

Reply:
I am sorry, this was really stupid :/